ClinGen/clincoded
# Connects to #1323. Connects to #1335. Connects to #1444. Connects to #1445. Connects to #1446. Connects to #1467. Connects to #1472. Connects to #1477. Connects to #1478.

**Steps to test #1323:**
1. In a new Experimental Data curation, select either the "Model Systems" or the "Rescue" option.
2. Then select "Non-human model organism" for either the "Non-human model organism or cell culture model?" or the "Rescue observed in human, non-human model organism, cell culture model, or patient cells?" questions.
3. Then in the "Non-human model organism" selection dropdown, expect to see the 3 new options:
_Budding yeast (Saccharomyces cerevisiae) 4932
Chlamydomonas (Chlamydomonas reinhardtii) 3055
Fission yeast (Schizosaccharomyces pombe) 4896_
4. Select either one of these 3 new options and expect to be able to save the curation successfully.

***

**Steps to test #1335 (Part 1):**
1. In a new Experimental Data curation, select the "Rescue" option.
2. Then in the "Rescue observed in human, non-human model organism, cell culture model, or patient cells?" selection dropdown, expect to see the following 4 new options replacing the previous 2:
_Human
Non-human model organism
Cell culture model
Patient cells_
3. Select "Human" from the dropdown and expect to see the "Does patient variant rescue?" question and its checkbox being removed from the display.
4. Select "Patient cells" from the dropdown and expect to see the "Does patient variant rescue?" question and its checkbox being shown again.
5. Fill out the other required fields and save the curation. Expect to be able to complete the form submission successfully.

**Steps to test #1335 (Part 2):**
1. Compare the following evidence in both production and test instances:
(Patient cells) https://curation.clinicalgenome.org/experimental/da959627-c119-47d1-b5b6-a5982f51ebee/
(Patient cells) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/da959627-c119-47d1-b5b6-a5982f51ebee/
(Engineered equivalent) https://curation.clinicalgenome.org/experimental/c9ea4b30-fe92-4ef2-b698-175abd431a56/
(Cell culture model) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/c9ea4b30-fe92-4ef2-b698-175abd431a56/
2. In the "Rescue" panel, expect to see the labels and user-entered values are matched or mapped correctly.

***

**Steps to test #1444 (Part 1):**
1. In a new Experimental Data curation, select the "Model Systems" option.
2. Then in the "Non-human model organism or cell culture model?" selection dropdown, expect to see the following 2 new options replacing the previous 2:
_Non-human model organism
Cell culture model_
3. Select "Cell culture model" from the dropdown and fill out the other required fields. Make sure to score this evidence.
4. Save the curation, and expect to be able to complete the form submission successfully.

**Steps to test #1444 (Part 2):**
1. Compare the following evidence in both production and test instances:
(Animal model) https://curation.clinicalgenome.org/experimental/63c9f2fa-b07d-49e4-b551-50300e50458f/
(Non-human model organism) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/63c9f2fa-b07d-49e4-b551-50300e50458f/
(Engineered equivalent) https://curation.clinicalgenome.org/experimental/bb51e417-79ef-4fdb-9134-d46fef0fd759/
(Cell culture model) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/bb51e417-79ef-4fdb-9134-d46fef0fd759/
2. In the "Model Systems" panel, expect to see the labels and user-entered values are matched or mapped correctly.

***

**Steps to test #1446 (Part 1):**
1. In a new Experimental Data curation, select the "Functional Alteration" option.
2. Then in the "Cultured patient or non-patient cells carrying candidate variant(s)?" selection dropdown, expect to see the following 2 new options replacing the previous 2:
_Patient cells
Non-patient cells_
3. Select "Non-patient cells" from the dropdown and fill out the other required fields. Make sure to score this evidence.
4. Save the curation, and expect to be able to complete the form submission successfully.

**Steps to test #1446 (Part 2):**
1. Compare the following evidence in both production and test instances:
(Patient cells) https://curation.clinicalgenome.org/experimental/fa2ae5bd-d5df-4f4d-a477-656c46dbcfb4/
(Patient cells) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/63c9f2fa-b07d-49e4-b551-50300e50458f/
(Engineered equivalent) https://curation.clinicalgenome.org/experimental/e56bd402-b38c-4d91-9a50-95a75a5b9f28/
(Non-patient cells) https://1335-experimental-prod-data.demo.clinicalgenome.org/experimental/e56bd402-b38c-4d91-9a50-95a75a5b9f28/
2. In the "Model Systems" panel, expect to see the labels and user-entered values are matched or mapped correctly.

***

**Steps to test #1445:**
1. Given the 3 previously saved/scored Experimental Data evidence, click on the "Generate Summary" button at the top right hand corner of the Curation-Central page.
2. Upon landing on the Score Matrix page, expect to see the evidence types of the 3 Experimental sections (e.g. Functional Alteration, Model Systems and Rescue) are properly named and displayed according to https://github.com/ClinGen/clincoded/issues/1445#issuecomment-328959409.
3. Also expect to see their scores being correctly calculated.

***

**Steps to test #1467:**
1. From the curation palette on the Curation-Central page, click on the "View/Score" or "Edit" links for the previously saved/scored "Functional Alteration" or "Model Systems" Experimental Data evidence.
2. Expect to be able to see and edit the previously saved score in the evidence.
3. The Biochemical Function's option B in Experimental Data is another example of this test case.

***

**Steps to test #1472:**
1. In one of the previously saved Experimental evidence, add a variant by CA ID and save the evidence.
2. From the curation palette on the Curation-Central page, click on the "View/Score" or "Edit" links for this evidence.
3. Expect to be able to see or edit the evidence with the CA ID variant.

***

**Steps to test #1477:**
1. On the Curation-Central page, expect to see the removal of the (GDM) status displayed in the static header.

***

**Steps to test #1478:**
1. Given the previously saved "Model Systems" Experimental evidence, go back and edit the curation form.
2. In the "Cell culture model type/line (EFO or CL ID):" text input field, remove the existing ID(s) if there's any.
3. Add an incorrect non-EFO/CL ID, or just some random text, into this text field. Click the "Save" button.
4. Expect to see an error message shown right beneath the text field (in addition to the one next to the buttons at the bottom) and the text field itself being highlighted as well.
